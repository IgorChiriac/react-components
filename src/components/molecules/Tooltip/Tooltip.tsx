import React from 'react';
import Tippy, { TippyProps } from '@tippy.js/react';
import TooltipStyles from './TooltipStyles/TooltipStyles';

Tooltip.Styles = TooltipStyles;

export default function Tooltip({ children, ...config }: Omit<TippyProps, 'animation' | 'theme' | 'flipOnUpdate'>) {
  return (
    <Tippy allowHTML={false} placement="top" animation="shift-toward" flipOnUpdate={true} inertia={true} {...config}>
      {children}
    </Tippy>
  );
}
