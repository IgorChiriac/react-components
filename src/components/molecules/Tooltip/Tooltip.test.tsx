import { axe } from 'jest-axe';
import React from 'react';
import { render } from '@testing-library/react';
import Tooltip from './Tooltip';

describe('<Tooltip />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container } = render(
      <Tooltip content="I prefer 🍕 over 🍰 ...">
        <span>💁🏻‍♂️</span>
      </Tooltip>,
    );

    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
