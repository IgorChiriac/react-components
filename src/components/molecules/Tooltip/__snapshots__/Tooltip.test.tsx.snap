// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders the component with no a11y violations 1`] = `
<span>
  💁🏻‍♂️
</span>
`;
